todo-list-app/src/utils/dragDrop.js

```diff
@@ -0,0 +1,229 @@
+import {EventEmitter} from './event';
+import {querySelector} from './helper';
+
+export class DragAndDrop {
+  // #region public field
+  // 드래그 요소
+  targetElement = null;
+
+  // 드랍 요소
+  dropzoneElement = null;
+
+  // 커스텀 데이터
+  customData = {};
+  // #endregion
+
+  // #region private field
+  // 타켓 좌측으로부터 위치
+  #diffX = 0;
+
+  // 타겟 상단으로부터 위치
+  #diffY = 0;
+
+  // 드래그 실행 중 여부
+  #isDragging = false;
+
+  // 드롭존인지 여부
+  #isDropzone = false;
+
+  // 원본 스타일
+  originStyles = {};
+  // #endregion
+
+  // #region getter
+  // 이벤트 파라메타
+  get dragEventParameter() {
+    return {
+      target: this.targetElement,
+      dropTarget: this.dropzoneElement,
+      customData: this.customData
+    };
+  }
+  // #endregion
+
+  // #region 메서드
+  /**
+   * 드래그 사용 할 요소 등록
+   *
+   * @param {string | HTMLElement} selector 드래그 요소 선택자
+   * @param {any} customData 사용자 데이터
+   */
+  draggable(selector, customData) {
+    // 드래그 요소가 등록되어 있을 경우
+    if (this.targetElement) {
+      return;
+    }
+
+    const targetElement = querySelector(selector);
+
+    // 요서 탐색했을 때 없을 경우
+    if (!targetElement) {
+      return;
+    }
+
+    // 속성 적용
+    this.targetElement = targetElement;
+    this.customData = customData;
+    // 요소 클릭 상태에서 마우스 움직 일 경우 핸들러
+    this.mousemoveHandler = this.mousemoveHandler.bind(this);
+    this.mouseupHandler = this.mouseupHandler.bind(this);
+    this.keyupHandler = this.keyupHandler.bind(this);
+
+    targetElement.addEventListener('mousedown', ({target, clientX, clientY}) => {
+      const {dropzoneElement} = this;
+      // 타겟 요소의 원본 스타일 저장
+      const {top, left, zIndex, userSelect, position} = targetElement.style;
+      this.originStyles.top = top;
+      this.originStyles.left = left;
+      this.originStyles.zIndex = zIndex;
+      this.originStyles.userSelect = userSelect;
+      this.originStyles.position = position;
```

Comment:
```suggestion
      const {top, left, zIndex, userSelect, position} = targetElement.style;
      this.originStyles = {top, left, zIndex, userSelect, position};
```
이렇게 줄일 수 있겠네요. originStyles에 이 외의 값은 저장을 하지 않는 것 같아서요.

Reply:
[efe7f87](https://github.com/JSero/nhn-cl-training/pull/5/commits/efe7f875d7860fa18d0ea6b5e9806322fa981e6a)

넵~ 수정했습니다!